lib/report.js

```diff
@@ -314,7 +307,7 @@ class TapReporter extends Reporter {
     if (test.success) {
       this.successful++;
     } else if (!test.todo) {
-      this.failed++;
+      this.failedTests.push(this.testNumeration);
```

Comment:
[Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) does not have `push` method.